dotCMS/src/main/java/com/dotcms/filters/interceptor/meta/MetaWebInterceptor.java

```diff
@@ -0,0 +1,47 @@
+package com.dotcms.filters.interceptor.meta;
+
+import com.dotcms.enterprise.ClusterUtilProxy;
+import com.dotcms.filters.interceptor.Result;
+import com.dotcms.filters.interceptor.WebInterceptor;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.StringUtils;
+import com.liferay.util.StringPool;
+import io.vavr.Lazy;
+import io.vavr.control.Try;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Interceptor that adds meta info to the dotcms response
+ * @author jsanca
+ */
+public class MetaWebInterceptor implements WebInterceptor {
+
+    private static final String X_DOT_SERVER_HEADER = "x-dot-server";
+    private static final String FRIENDLY_NAME = "friendlyName";
+    private static final String UNKNOWN = "unknown";
+    private final Lazy<Boolean> responseHeaderAddNodeId = Lazy.of(()-> Config.getBooleanProperty("RESPONSE_HEADER_ADD_NODE_ID", true));
+    private final Lazy<String>  serverId                = Lazy.of(()-> StringUtils.shortify(APILocator.getServerAPI().readServerId(), 10));
+    private final Lazy<Map<String, Serializable>> cacheInfoMap = Lazy.of(() -> Try.of(() -> ClusterUtilProxy.getNodeInfo()).getOrElse(Collections.emptyMap()));
```

Comment:
I would just cache the `FriendlyName` as a String and not the whole NodeMap, which can get big.  Also, the inclusion of the `FriendlyName` should be configurable.

Reply:
ok